### ISSUE TITLE
Stat: No such file or directory

### ISSUE THREAD

**Issue by Nutomic** [syncthing/syncthing#1253]
I'm getting this for a bunch of files and folders (which I probably deleted while syncthing wasn't running):
```
[UE6TV] 11:42:05 INFO: Puller (folder "x", file "path/to/file"): dst stat dir: stat /repo/path no such file or directory
...
[UE6TV] 11:42:05 WARNING: Folder "x" isn't making progress - check logs for possible root cause. Pausing puller for 1m0s.
```

However, there's no apparent way to fix this (short of recreating the files and then deleting them again).

**Comment by AudriusButkevicius** [comment]
Can you enable some random STTRACE flag?
That starts showing the line number of where this is happening.

**Comment by Nutomic** [comment]
[UE6TV] 2015/01/14 11:52:13.025758 sharedpullerstate.go:172: INFO: Puller (folder "x", file "y"): dst stat dir: stat z: no such file or directory

**Comment by AudriusButkevicius** [comment]
When did this start happening or was this always an issue?
Is there a way to reliable reproduce this?

It seems that the directory we want to create the temp file in does not exist, but I am not sure why the directory creation hasn't happened,

**Comment by Nutomic** [comment]
I've noticed it for the first time now.

I also tried to reproduce it, but can't. Maybe this is due to some weird edge case I had, so feel free to close it.

Btw deleting the folder in question on the other side fixed the problem.

**Comment by bigbear2nd** [comment]
Btw deleting the folder in question on the other side fixed the problem

I have a similar issue when i sync files which change quickly. For me, a rescan of the remote folder fixed the problem.

**Comment by calmh** [comment]
Can we have have a log line without x, y and z? ROT13 it if you like. :) Specifically I'm interested in what the path is to the file as relative to the folder root...

**Comment by Nutomic** [comment]
The path withiin the repo was like `folderA/folderB/file`, if it still helps ;)

**Comment by seidler2547** [comment]
I have thousands of these log messages flooding my output. It's not a recent phenomenon, I think some time the puller got rewritten and then these messages started popping up. Here are some:

```
16:22:51 INFO: Puller (folder "NITSCHE-PC", dir "RECYCLER/S-1-5-21-584516674-491292040-2915802060-1006/Dc70/readmes"): mkdir /media/data/Backups/PC-NITSCHE3/RECYCLER/S-1-5-21-584516674-491292040-2915802060-1006/Dc70/readmes: no such file or directory
16:22:51 INFO: Puller (folder "NITSCHE-PC", dir "RECYCLER/S-1-5-21-584516674-491292040-2915802060-1006/Dc70"): mkdir /media/data/Backups/PC-NITSCHE3/RECYCLER/S-1-5-21-584516674-491292040-2915802060-1006/Dc70: no such file or directory
```
and so on. Sometimes about 50 per second of them. I'm backing up a Windows PC. Yes i should exclude RECYCLER, but until I get to the remote PC again, I can't edit the stignore file. Also, this also happens with other files in Documents and Settings.

**Comment by AudriusButkevicius** [comment]
Can you actually post more of the lines? Because it seems that the creations are happening out of order.

**Comment by seidler2547** [comment]
http://pastebin.com/KKiU4p5A

**Comment by AudriusButkevicius** [comment]
I looked at this, but it doesn't make a lot of sense. 
As a starting point, I suggest you reset the index on the device that is having these issues to see if it helps, also check for additional errors, and whether or not the directory there (up to the point it starts failing) has been built up.

If it persists, open a separate issue, as this one as it's failing at a different place, and might not be related.

**Comment by seidler2547** [comment]
How would I reset the index?

**Comment by AudriusButkevicius** [comment]
Call `syncthing -reset`

**Comment by seidler2547** [comment]
But this will remove the config, right? I mean, do I need to set up all shared folders again? And will the device ID change?

**Comment by AudriusButkevicius** [comment]
No, this will only remove the cached view of the directory tree.
Ideally, don't remove the config dir, just the delete the index directory.

**Comment by seidler2547** [comment]
WTF - `syncthing -reset` just renamed all my data folders????

**Comment by seidler2547** [comment]
I hope after deleting the index directory it will not think that all the data in the original folders does not belong but should be deleted? Can someone confirm before I trash a couple 100s of GB?

**Comment by AudriusButkevicius** [comment]
:scream: confused `-reset` with index reset. Yeah just seems to move the folders out of the way.
Removing the index will force the directories to be rescanned, which will cause high CPU usage until the data is indexed. Regardless, we should stop spamming this ticket, as it has nothing todo with your problem.

**Comment by AudriusButkevicius** [comment]
@Nutomic were you using inotify when this happened?
I feel this could be https://github.com/syncthing/syncthing-inotify/issues/22

**Comment by Nutomic** [comment]
I don't think I've used any kind of inotify for this folder, but I'm not sure anymore.

**Comment by calmh** [comment]
We solved a few similar things lately, and there is the known issue with inotify that we'll solve soon enough, so I'm going to go ahead and close this. Please yell if it happens again without inotify.

**Comment by xamindar** [comment]
This problem happened to me last night. I added a folder on all my devices that already had everything in the folders. 

Then when I started deleting some stuff on one machine, this error popped up on all my machines with this folder syncing. I had to delete it from them all machines, clean it up, then re-add it for it to work.

I don't see any inotify setting in the options.

**Comment by Nairou** [comment]
I just started using syncthing yesterday, and am seeing this problem today.

I setup a sync between two computers (A and B for clarity), for a "Pictures" folder containing thousands of photos in multiple folders. Both computers already had the full set of files, as they were previously sync'd by Bittorrent Sync.

Around this same time, I decided to do some housecleaning and deleted a few of the subfolders within the "Pictures" folder on computer A (the one I was physically using). Afterwards I noticed thatboth computers said they were disconnected (in the web GUI), so I told syncthing to restart (via the web GUI) on computer B, to see if that would reconnect them. It did, but then computer B proceeded to start copying the files I deleted back to computer A. It apparently didn't know that the deletions were intentional. It was going to transfer a lot of files I didn't want transferred and I didn't know how to stop it, so I logged in to computer B and deleted those subfolders there as well.

Shortly after that, I started getting "Folder 'Pictures' isn't making progress" messages on both computers. I checked the syncthing output on computer B and saw "Stat: No such file or directory" errors about the subfolders I deleted.
